File: api/removebg.js
import { removeBackground } from "rembg-node";

export default async function handler(req, res) {
  try {
    if (req.method !== "POST") {
      return res.status(405).json({ error: "Only POST allowed" });
    }

    const chunks = []
    for await (const chunk of req) {
      chunks.push(chunk)
    }
    const buffer = Buffer.concat(chunks)

    if (!buffer || buffer.length < 20) {
      return res.status(400).json({ error: "No image provided" })
    }

    const output = await removeBackground(buffer)

    res.setHeader("Content-Type", "image/png")
    res.send(output)
  } catch (err) {
    res.status(500).json({ error: "Processing failed" })
  }
}

File: api/removebg.py
from rembg import remove
from fastapi import FastAPI, Request
from fastapi.responses import Response

app = FastAPI()

@app.post("/api/removebg")
async def remove_bg(req: Request):
    img_bytes = await req.body()
    if not img_bytes:
        return {"error": "No image provided"}

    output = remove(img_bytes)
    return Response(content=output, media_type="image/png")
